src/js/utils/date-utils.js

```diff
@@ -0,0 +1,283 @@
+/**
+ * Create a formatted, locale-aware string representation of a datetime.
```

Comment:
It would be helpful to add an example of how a client would call the function(s) in this module.